processor/src/memory.rs

```diff
@@ -0,0 +1,11 @@
+use std::collections::{BTreeMap, HashMap};
+use crate::GoldilocksField;
+
+#[derive(Debug, Default)]
+pub struct Memory {
+    // store current memory state
+    pub state: HashMap<u64, GoldilocksField>,
+
+    // visit by address, (u32, u32) store op clk and memory value
+    pub trace: BTreeMap<u64, Vec<(u32, GoldilocksField)>>,
```

Comment:
The whole memory trace should be:
| address | clk | pc | value | rw |
| -------- | -------- | -------- | -------- | -------- |

and perhaps address should be u32?

Reply:
Because register store GoldilocksField, It use GoldilocksField as address also ok.  When need use address as u32, can convert into u32 feasibly.